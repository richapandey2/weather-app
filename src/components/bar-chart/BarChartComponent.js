import React from "react";
import { Dialog, DialogContent, Slide } from "@material-ui/core";
import PropTypes from "prop-types";
import { VictoryChart, VictoryAxis, VictoryBar, Bar } from "victory";
import { convertFahrenheittoCelcius } from "../../utils/formatter";
import {
  TEMP_FAHRENHEIT_LABEL,
  TEMP_CELCIUS_LABEL,
  TIME,
} from "../../constants/Constants";

function BarChartComponent({ chartDetails, handleDetailsDialog, isCelcius }) {
  const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />;
  });
  return (
    <Dialog
      open
      TransitionComponent={Transition}
      onClose={() => handleDetailsDialog(false)}
    >
      <DialogContent>
        <VictoryChart
          height={400}
          width={500}
          domainPadding={{ x: 50, y: [0, 20] }}
        >
          <VictoryAxis
            dependentAxis
            offsetX={50}
            style={{ axisLabel: { padding: 35 } }}
            label={isCelcius ? TEMP_CELCIUS_LABEL : TEMP_FAHRENHEIT_LABEL}
          />
          <VictoryAxis label={TIME} />
          <VictoryBar
            barWidth={30}
            dataComponent={<Bar />}
            style={{
              data: { fill: "#3f51b5" },
            }}
            data={chartDetails}
            x="time"
            y="temp"
            labels={({ datum }) => {
              return isCelcius
                ? convertFahrenheittoCelcius(datum.temp)
                : datum.temp;
            }}
          />
        </VictoryChart>
      </DialogContent>
    </Dialog>
  );
}

BarChartComponent.propTypes = {
  chartDetails: PropTypes.arrayOf(
    PropTypes.shape({
      time: PropTypes.string,
      temp: PropTypes.number,
    })
  ),
  handleDetailsDialog: PropTypes.func,
  isCelcius: PropTypes.bool,
};

export default BarChartComponent;
