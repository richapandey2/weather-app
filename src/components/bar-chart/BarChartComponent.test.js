import React from "react";
import { render, cleanup } from "@testing-library/react";
import BarChartComponent from "./BarChartComponent";

afterEach(cleanup);

test("should test bar chart component", () => {
  const chartDetail = [{ time: "00:00", temp: 20 }];
  const { asFragment } = render(
    <BarChartComponent chartDetails={chartDetail} />
  );
  expect(asFragment).toMatchSnapshot();
});
