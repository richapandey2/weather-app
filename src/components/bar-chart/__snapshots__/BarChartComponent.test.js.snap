// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test bar chart component 1`] = `[Function]`;
